src/darwin/Framework/CHIP/MTRAsyncWorkQueue.mm

```diff
@@ -29,6 +30,25 @@ typedef NS_ENUM(NSInteger, MTRAsyncWorkItemState) {
     MTRAsyncWorkItemRetryCountBase = MTRAsyncWorkItemRunning, // values >= MTRAsyncWorkItemRunning encode retryCount
 };
 
+// A helper struct that facilitates access to _context while
+//  - only reading the _context weak reference once and retaining a strong
+//    reference for the duration of a particular queue method call
+//  - avoiding calls to `[context description]` under our lock
+struct ContextSnapshot {
+    id _Nullable reference;
+    NSString * description;
+    ContextSnapshot(id _Nullable context)
+    {
+        reference = context;
+        description = [context description] ?: @"?";
+    }
+};
+
+MTR_DIRECT_MEMBERS
+@interface MTRAsyncWorkItem ()
+@property (readonly, nonatomic) NSUInteger uniqueID;
```

Comment:
I'm not sure that's a realistic problem... even if it did overflow it would take long enough that you can easily distinguish the tasks with reused IDs in the logs.

Reply:
That's fair, but what is the drawback of using uint64_t here?

Reply:
I guess there's no specific drawback... The strong default in ObjC is just to use `NS(U)Integer` but I'll change it to uint64_t.